test/xUnit/csharp/test_CompletionHelpers.cs

```diff
@@ -8,6 +8,24 @@ namespace PSTests.Parallel
 {
     public class CompletionHelpersTests
     {
+        [Theory]
+        [InlineData("word", "'", "'word'")]
+        [InlineData("word", "\"", "\"word\"")]
+        [InlineData("word's", "'", "'word''s'")]
+        [InlineData("already 'quoted'", "'", "'already ''quoted'''")]
+        [InlineData("", "'", "''")]
+        [InlineData("", "\"", "\"\"")]
+        [InlineData("'", "'", "''''")]
+        [InlineData("", "", "''")]
```

Comment:
I still think we should cover CompletionRequiresQuotes by tests too. There we use Parser and additionally checks for backtick and `$`.
So string for tests could be like "a b", "$a", "`a"
Maybe you could think more tests - I see `StringToken` and `TokenFlags.Keyword` in the method. The test strings could be "a$a" and "while" accordingly.

Reply:
@iSazonov Added some tests here for CompletionRequiresQuotes: fced8d73032763f89f7167c8a34a73aba540d239

Reply:
Thanks!

The QuoteCompletionText is more interesting for us. Maybe add 2-3 key tests here?

Reply:
Up!

Reply:
@iSazonov Ok I have added more tests and reorganized them in 9bff47bcc099506d0e3fca35d308b8465e6ae0cd.